### DISCUSSION TITLE
psycopg2.errors.OutOfMemory: out of memory

### DISCUSSION THREAD

**Discussion by akki401** [mediagis/nominatim-docker#562] [General]
Hi I started docker for europe planet file (europe-latest.osm.pbf) which gives the psycopg2.errors.OutOfMemory: out of memory error.
After all finishing the RANK 30, the docker started again from start and start indexing.

**EC2 config:**
i4i.2xlarge
vCPU = 8
Memory (GiB) = 64
Instance Storage (GB) = 1 x 1,875 AWS Nitro SSD
Network Bandwidth (Gbps) = Up to 12
EBS Bandwidth (Gbps) = Up to 10

**Docker command:**
        cmd = " ".join(
            [
                "docker",
                "run",
                "--restart=always",
                "--log-driver json-file",
                "--log-opt max-size=10m",
                "--log-opt max-file=3",
                "--shm-size=4g",
                "-e",
                "POSTGRES_SHARED_BUFFERS=2GB",
                "-e",
                "POSTGRES_MAINTAINENCE_WORK_MEM=10GB",
                "-e",
                "POSTGRES_AUTOVACUUM_WORK_MEM=2GB",
                "-e",
                "POSTGRES_WORK_MEM=50MB",
                "-e",
                "POSTGRES_EFFECTIVE_CACHE_SIZE=24GB",
                "-e",
                "POSTGRES_SYNCHRONOUS_COMMIT=off",
                "-e",
                "POSTGRES_MAX_WAL_SIZE=1GB",
                "-e",
                "POSTGRES_CHECKPOINT_TIMEOUT=10min",
                "-e",
                "POSTGRES_CHECKPOINT_COMPLETITION_TARGET=0.9",
                "-e",
                f"PBF_PATH=/nominatim/data/{self.region}-latest.osm.pbf",
                "-e",
                f"REPLICATION_URL={self._updates_url}",  # source url of the updates for this area
                "-e",
                f"REPLICATION_UPDATE_INTERVAL={3600 * 24 * 30}",  # How often upstream publishes diffs
                # (in seconds, default: 86400).# update every 30 days
                "-e",
                f"REPLICATION_RECHECK_INTERVAL={3600 * 24 * 15}",  # How long to sleep if no update found yet
                # (in seconds, default: 900).# sleep 15 days
                "-e",
                "UPDATE_MODE=none",  # continuous/once/catch-up/none
                "-e",
                "FREEZE=false",
                "-e",
                "IMPORT_STYLE=full",
                "-e",
                "NOMINATIM_FLATNODE_FILE=/nominatim/flatnode/flatnode.file",
                "-v",
                f"{self._dir_nominatim_postgressql_db}/:/var/lib/postgresql/14/main",
                "-v",
                f"{self._dir_nominatim}/:/nominatim/data",
                "-v",
                f"{self._dir_nominatim_flatnode}/:/nominatim/flatnode",
                "-p",
                f"{self._nominatim_port}:8080",
                "-i",
                "-t",
                "-d",
                "--name",
                self._nominatim_container_name,
                "mediagis/nominatim:4.4",
            ]
        )

I felt the system configuration is good enough in terms of RAM and storage to run the europe continent nominatim
Let me know if anything need to change in the docker config command which I posted above

Disk space at 26May 2024 6:52 PM IST
![image](https://github.com/mediagis/nominatim-docker/assets/50397166/a9f51d2b-2adc-4df7-a1a3-c4984f344940)

**Comment by leonardehrenfried** [comment]
What do you think "OutOfMemory: out of memory" could possibly mean?

**Reply by akki401** [reply]
Out of memory could be possible due to lack of RAM memory?
But 64GB is not enough to do indexing the europe continent?

**Reply by akki401** [reply]
As of now (May 27th 2024 12 PM IST) the memory usage is
![image](https://github.com/mediagis/nominatim-docker/assets/50397166/0636f272-9ed3-4c5e-bb3c-fea92d9b2910)

**Reply by mtmail** [reply]
>  But 64GB is not enough to do indexing the europe continent?

Apparently not.
